### PR TITLE
Adjust MIN_PERP_DIST & MAX_LANE_DOT_PRODUCT to avoid Starlane Nexus c…

### PR DESCRIPTION
**Pull request by Oberlus**
…rossing over systems

Forum thread: https://www.freeorion.org/forum/viewtopic.php?f=15&t=9719

### CODE REVIEW

**Comment by Oberlus** [comment]
This works fine for the original purpose (avoid starlanes created by Starlane Nexus to cross over systems or be too close to other starlanes).

However, it might be creating a new problem:
I've started 10 games with random seed, 150 systems, 11 players, high starlane frequency, to check if the proposed changes affect universe generation.
The first time the server exited (crashed?) unexpectedly. I couldn't see the reason in the logs, so I don't know if it was caused by the committed changes or by something else unrelated.
The other nine times worked as expected and the universes generated seem just fine.
So I don't know if this can be merged or not. I would appreciate someone with more insight to check the attached logs for the one game that exited after universe generation.

[freeorion.log](https://github.com/freeorion/freeorion/files/3808932/freeorion.log)
[freeoriond.log](https://github.com/freeorion/freeorion/files/3808933/freeoriond.log)

**Comment by Oberlus** [comment]
Started another 10 games with 300 systems, starlane high, 20 players. No problems so far.

**Comment by Oberlus** [comment]
Looks like the AppVeyor build failed due to excessive time (1h). Should I do anything about that?

**Comment by Vezzra** [comment]
> Looks like the AppVeyor build failed due to excessive time (1h). Should I do anything about that?

No.

File: universe/Condition.cpp [hunk 1]
```diff
@@ -7511,7 +7511,7 @@ namespace {
         dy2 /= mag;
 
 
-        const float MAX_LANE_DOT_PRODUCT = 0.98f;   // magic limit copied from CullAngularlyTooCloseLanes in UniverseGenerator
+        const float MAX_LANE_DOT_PRODUCT = 0.87f;   // magic limit copied from CullAngularlyTooCloseLanes in UniverseGenerator
```
Comment (geoffthemedio) [inline universe/Condition.cpp]: update comment to remove reference to https://github.com/freeorion/freeorion/blob/master/universe/UniverseGenerator.cpp#L743

Comment (Oberlus) [reply universe/Condition.cpp]: Hopefully fixed

**Pull request merged**